SimModel.hpp

```diff
@@ -37,25 +56,473 @@ namespace SIMMODEL
+        
+        double Ue = body->LewisConstant/(
+            njActual*body->Icl/body->icl
+            +1/(f*hcmix)
+        );
+        double dsweat = elem->a_sw*sect->areaShare*Sw*pow(2,(TsfAssumed-Tsf0)*transient/Q10) * 1e-3/60;
+        double Psk = (body->LambdaH20*dsweat + Posksat*body->Rinverse + Ue*Pair)/
+                (Ue + body->Rinverse);
+        double ans = U*(TsfAbs-Tweight) + Ue*(Psk-Pair);
+        assert(!isnan(abs(ans)));
+        return ans;
+    }
 
     SimModelState SimModel::getState()
     {
         return _state;
     }
-    class KonstasCase : SimModel
+    class InitialCase : public SimModel
     {
+        public:
+            InitialCase() : SimModel() {
+                thermovariant = 0;
+                transient = 0;
+                tFinal = 1;
+                dt = 1;
+            };
 
     };
+    class SteadyCase : public SimModel
+    {
+        public:
+            SteadyCase() : SimModel() {
+                thermovariant = 1;
+                transient = 0;
+                tFinal = 10;
+                dt = 1;
+            };
+    };
+    class TransientCase : public SimModel
+    {
+        public:
+            TransientCase() : SimModel() {
+                thermovariant = 1;
+                transient = 1;
+                tFinal = 20.0;
+                dt = .1;
+            };
+    };
+    class InjuryCase : public SimModel
+    {
+        public:
+            double tinjury = 300; // s
+            double bvrRate = -.4/tinjury; // m^3/m^3 / s
+            InjuryCase() : SimModel() {
+                thermovariant = 1;
+                transient = 1;
+                tFinal = 3600.0;
+                dt = 0.1;
+            };
+            void BVRSVR( double *bvr, double *svr, double time ) override
+            {
+                if(time <= tinjury){
+                    *bvr = 1.0 + bvrRate*(time);
+                    *svr = 0;
+                }else{
+                    *bvr = 1.0 + bvrRate*(tinjury);
+                    *svr = 0;
+                }
+            };
+            bool endCondition(double Thy)
+            {
+                return Thy<=10+273.15;
+            };
+    };
+    class KonstasCase : public SimModel
+    {
+        public:
+            double tinjury = 300; // s
+            double bvrRate = -.4/tinjury; // m^3/m^3 / s
+            double trestore = 120; // s, time to restore blood volume
+            KonstasCase() : SimModel() {
+                thermovariant = 1;
+                transient = 1;
+                tFinal = 3600.0;
+                dt = 1.0;
+            };
+            void rhocp(
+                double *Rho,
+                double *Cp,
+                BODYMODEL::BodyModel *body,
+                int eleIdx,
+                double Viv,
+                double DViv,
+                double fEB,
+                double fES,
+                double time
+            ) override {
+                if(eleIdx<3 && time>=trecovery){
```

Comment:
I don't think we need *lots* of commenting in these parts of the code, but I am curious what this step is doing and perhaps some quick comments could shed light on the next couple calculations for Rho, Cp, etc

Reply:
Can do. I should put a reference to the equation used. It's a weighting of incoming ECMO flow and the existing saline/blood pool.